geopandas/io/sql.py

```diff
@@ -156,6 +173,14 @@ def _read_postgis(
     >>> df = geopandas.read_postgis(sql, con)  # doctest: +SKIP
     """
 
+    try:
+        spatial_ref_sys_sql = "SELECT * FROM spatial_ref_sys"
+        spatial_ref_sys_df = (
+            pd.read_sql(spatial_ref_sys_sql, con) if crs is None else None
+        )
```

Comment:
Do you have an idea how large this table typically is? Because we are now reading this full table every time when calling `read_postgis`, and _ideally_ we would already do the filtering step here in the query, instead of doing it on the dataframe. I know we don't know the `srid` value at this point for specifying it in the query, but I wonder if we could maybe pass `con` to `_df_to_geodf` and then do this query there with directly filtering on srid. We would then only have to "cache" this somehow for repeated calls to `_df_to_geodf` from a single `read_postgis` call.

Reply:
Mine is 8000 rows with approx 7.3mb

```
postgres=# SELECT pg_size_pretty( pg_total_relation_size('spatial_ref_sys') );
 pg_size_pretty 
----------------
 7280 kB
(1 row)
```

Reply:
That's a decent chunk of data to have to transfer each time. I would try to look into querying only the one `srid` value that is needed.